source/guides/references/migration-guide.md

```diff
@@ -0,0 +1,299 @@
+
+```javascript
+// Would pass in Cypress 3 but will fail in 4
+expect(true).to.be.ture
+```
+
+## Sinon.JS upgrade
+
+Sinon.JS was upgraded from `3.2.0` to `8.1.1`, which includes a number of breaking changes and new features outlined in {% url "Sinon.JS's migration guide" https://sinonjs.org/releases/latest/#migration-guides %}. Some changes you might notice are described below.
+
+### {% fa fa-warning red %} Breaking Change: stub non-existent properties
+
+An error will throw when trying to stub a non-existent property.
+
+```javascript
+// Would pass in Cypress 3 but will fail in 4
+cy.stub(obj, 'nonExistingProperty')
+```
+
+### {% fa fa-warning red %} Breaking Change: `reset()` replaced by `resetHistory()`
+
+For spies and stubs, the `reset()` method was replaced by `resetHistory()`.
+
+{% badge danger Before %} Spies and stubs using `reset()`.
+
+```javascript
+const spy = cy.spy()
+const stub = cy.stub()
+
+spy.reset()
+stub.reset()
+```
+
+{% badge success After %} Update spies and stubs should now use `resetHistory()`.
+
+```javascript
+const spy = cy.spy()
+const stub = cy.stub()
+
+spy.resetHistory()
+stub.resetHistory()
+```
+
+## Chromium based browser `family`
+
+We updated the {% url "Cypress browser objects" browser-launch-api %} of all Chromium-based browsers, including Electron, to have `chromium` set as their `family` field.
+
+```js
+module.exports = (on, config) => {
+  on('before:browser:launch', (browser = {}, args) => {
+    if (browser.family === 'electron') {
+      // would match Electron in 3.x
+      // will match no browsers in 4.0.0
+      return args
+    }
+
+    if (browser.family === 'chromium') {
+      // would match no browsers in 3.x
+      // will match any Chromium-based browser in 4.0.0
+      // ie Chrome, Canary, Chromium, Electron, Edge (Chromium-based)
+      return args
+    }
+  })
+}
+```
+
+{% badge danger Before %} This will no longer find the Electron browser.
+
+```js
+module.exports = (on, config) => {
+  on('before:browser:launch', (browser = {}, args) => {
+    if (browser.family === 'electron') {
+      // run code for Electron browser in 3.x
+      return args
+    }
+  })
+}
+```
+
+{% badge success After %} Use `browser.name` to check for Electron
+
+```js
+module.exports = (on, config) => {
+  on('before:browser:launch', (browser = {}, args) => {
+    if (browser.name === 'electron') {
+      // run code for Electron browser in 4.0.0
+      return args
+    }
+  })
+}
+```
+
+## `--browser` flag for release channels
+
+The {% url "`--browser` flag" command-line#cypress-run-browser-lt-browser-name-or-path-gt %} has been updated so you can specify a specific release channel of a browser to run.
+
+This means that release channels that could previously be passed as the sole argument need to be prepended with the browser name.
+
+{% badge danger Before %} This will no longer run Chrome Canary.
```

Comment:
we kept this in for backwards compatibility, so there is no breaking change here